### ISSUE TITLE
Prediction looks bad

### ISSUE THREAD

**Issue by Fang-Haoshu** [FengyangZhang/caffe_roomnet#5]
Hi,

After training the model, I visualize some output of the model and find that the estimated layout looks bad, even on training set. Here are some results. Does anybody successfully train the network and have good results?

![0823881db7b534c21fa5ffea961756f01f32ad0f](https://user-images.githubusercontent.com/12446953/29746069-cefa0644-8a82-11e7-833b-7f7a72497380.jpg)
![06f630a7d1a65ef4f9c56858bea29bede349d83a](https://user-images.githubusercontent.com/12446953/29746073-cefac41c-8a82-11e7-91d0-fcf60fd1da7e.jpg)
![0866893bd7a07060d7814aaac4fa1412829abc7b](https://user-images.githubusercontent.com/12446953/29746074-cefc84b4-8a82-11e7-93ab-6f73787f4914.jpg)
![08e3d8e9a6311a36d61e23d70e0cfe85d4b80a15](https://user-images.githubusercontent.com/12446953/29746071-cefa7d86-8a82-11e7-8e51-59268b2dbc1f.jpg)
![07cbdd41f4078a19ffde2d0f242f3adbf98ba6b9](https://user-images.githubusercontent.com/12446953/29746072-cefaab8a-8a82-11e7-8b50-e4702ef60d42.jpg)
![07e07792f5be5208f56fec3bb8c54047728d2497](https://user-images.githubusercontent.com/12446953/29746070-cefa834e-8a82-11e7-86d4-6f63eb376ae4.jpg)
![07be471d7616c6f2fe5dc6269536daa7e248bd48](https://user-images.githubusercontent.com/12446953/29746075-cf0f75a6-8a82-11e7-855a-e9b6fa06fb0e.jpg)

**Comment by FengyangZhang** [comment]
Sorry, as it's an out-of-interest project, I didn't have much time to revise it and make the output better, in fact I stopped when I get reasonable results, you can do some more revision, or apply the recurrent model in the paper to see if you can get better results.

**Comment by kwangHouzz** [comment]
Hi Haoshu,

Are these visualization of the training set or test set? BTW, my result looks bad either.

**Comment by Fang-Haoshu** [comment]
Hi, they are visualization on training set. So I think the network is under-fitting

**Comment by kwangHouzz** [comment]
Hi Haoshu,

Thanks for your response!

I think there are a few bugs in his code. For starters, he is not subtracting mean values and scaling the raw pixels. I have built another data layer based on the image_data_layer.cpp. My classification accuracy is pretty much the same with the paper's result. Are you still working on this project? What about your accuracy on test set?

Thanks,
Kevin

**Comment by Fang-Haoshu** [comment]
I made some modifications and now my classification branch is a little bit over-fitted. The accuracy is 0.99 on training set but only 0.7x on testing set. But the keypoint branch works well so my final accuracy is on-par with the paper.
How about you? And any suggestions on the classification branch?

**Comment by kwangHouzz** [comment]
I had similar results. In practice, I found my heatmap prediction kind of noisy. How do you generate a keypoint based on heatmap? I tried with simple argmax function, and it seemed to work better with Gaussian Blur and manually zeroing out part of unlikely region before argmax.

BTW, I actually just reached 0.8 accuracy by including data augmentation (mirror all training images). Have you tried that with vanilla network or recurrent one? Mine is still vanilla network.

Thanks!

**Comment by euivmar** [comment]
@Fang-Haoshu @kwangHouzz  Can you share your code please? (A pull request for example).  If you can share also your models weights I would be very grateful!

**Comment by DRAhmadFaraz** [comment]
@Fang-Haoshu @kwangHouzz  Can you share your modified code please? 

my email : faraz6313@gmail.com

I would be very grateful!

**Comment by jkstyle2** [comment]
@Fang-Haoshu @kwangHouzz Can you please share the code please?

My email : jaykaypark@gmail.com

it will be very helpful!!
